source/server/admin/admin_factory_context.h

```diff
@@ -0,0 +1,85 @@
+#pragma once
+
+#include "envoy/server/factory_context.h"
+#include "envoy/server/instance.h"
+
+#include "source/common/config/metadata.h"
+
+namespace Envoy {
+namespace Server {
+
+class AdminFactoryContext final : public Configuration::FactoryContext {
```

Comment:
Do we need to inherit from FactoryContext?  If not I'd love to start out the simple APIs here.

Reply:
If we do need to inherit from factory context is there a base impl we can use rather than reimplementing all this wrapper code?

Reply:
> Do we need to inherit from FactoryContext? If not I'd love to start out the simple APIs here.

IMO, yeah. The access logger factory require `ListenerAccessLogFactoryContext` which is direct base class of `FactoryContext` and is derived from `CommonFactoryContext`.

And if we break the inheritance relationship in the future and add a `serverFactoryContext` method to the `ListenerAccessLogFactoryContext`, it will become almost same with the `FactoryContext`.

So, I am inclined to use `FactoryContext` directly and remove `ListenerAccessLogFactoryContext` in next PR.

Reply:
> If we do need to inherit from factory context is there a base impl we can use rather than reimplementing all this wrapper code?

Yeah, there is a possible base impl in the listener_manager code base. But we cannot re-use it directly. Additional refactoring of modifications are still necessary.

Considering most of these wrapper methods will be remove in next 3~4 refactoring PRs, I didn't want to involve code changes of listener_manager code base in this patch.

Feel free to tell me if you think this a block point. :)